pkg/controller/volume/selinuxwarning/selinux_warning_controller.go

```diff
@@ -0,0 +1,507 @@
+	c := &Controller{
+		kubeClient:        kubeClient,
+		podLister:         podInformer.Lister(),
+		podIndexer:        podInformer.Informer().GetIndexer(),
+		podsSynced:        podInformer.Informer().HasSynced,
+		pvLister:          pvInformer.Lister(),
+		pvsSynced:         pvInformer.Informer().HasSynced,
+		pvcLister:         pvcInformer.Lister(),
+		pvcsSynced:        pvcInformer.Informer().HasSynced,
+		csiDriverLister:   csiDriverInformer.Lister(),
+		csiDriversSynced:  csiDriverInformer.Informer().HasSynced,
+		vpm:               &volume.VolumePluginMgr{},
+		seLinuxTranslator: volumeutil.NewSELinuxLabelTranslator(),
+
+		recorder: recorder,
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig[cache.ObjectName](
+			workqueue.DefaultTypedControllerRateLimiter[cache.ObjectName](),
+			workqueue.TypedRateLimitingQueueConfig[cache.ObjectName]{
+				Name: "selinux_warning",
+			},
+		),
+		labelCache:    volumecache.NewVolumeLabelCache(),
+		eventRecorder: recorder,
+	}
+
+	err := c.vpm.InitPlugins(plugins, prober, c)
+	if err != nil {
+		return nil, fmt.Errorf("could not initialize volume plugins for SELinux warning controller: %w", err)
+	}
+	csiTranslator := csitrans.New()
+	c.csiTranslator = csiTranslator
+	c.cmpm = csimigration.NewPluginManager(csiTranslator, utilfeature.DefaultFeatureGate)
+
+	err = common.AddPodPVCIndexerIfNotPresent(c.podIndexer)
+	if err != nil {
+		return nil, fmt.Errorf("could not initialize SELinux warning controller: %w", err)
+	}
+
+	logger := klog.FromContext(ctx)
+	_, err = podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.addPod(logger, obj) },
+		DeleteFunc: func(obj interface{}) { c.deletePod(logger, obj) },
+		// Not watching updates: Pod volumes and SecurityContext are immutable after creation
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = pvcInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.addPVC(logger, obj) },
+		UpdateFunc: func(oldObj, newObj interface{}) { c.updatePVC(logger, oldObj, newObj) },
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = pvInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.addPV(logger, obj) },
+		UpdateFunc: func(oldObj, newObj interface{}) { c.updatePV(logger, oldObj, newObj) },
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = csiDriverInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.addCSIDriver(logger, obj) },
+		UpdateFunc: func(oldObj, newObj interface{}) { c.updateCSIDriver(logger, oldObj, newObj) },
+		DeleteFunc: func(obj interface{}) { c.deleteCSIDriver(logger, obj) },
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	RegisterMetrics(logger, c.labelCache)
+
+	return c, nil
+}
+
+func (c *Controller) addPod(logger klog.Logger, obj interface{}) {
```

Comment:
should we fix these logger parameters to `_` if we are not using them to prevent linter warnings?